db/engine/upn.service.js

```diff
@@ -0,0 +1,277 @@
+'use strict'
+
+/** @namespace */
+
+const remainderLookup = {
+  0: 'A',
+  1: 'B',
+  2: 'C',
+  3: 'D',
+  4: 'E',
+  5: 'F',
+  6: 'G',
+  7: 'H',
+  8: 'J',
+  9: 'K',
+  10: 'L',
+  11: 'M',
+  12: 'N',
+  13: 'P',
+  14: 'Q',
+  15: 'R',
+  16: 'T',
+  17: 'U',
+  18: 'V',
+  19: 'W',
+  20: 'X',
+  21: 'Y',
+  22: 'Z'
+}
+
+// We want these LA codes to be unique
+const validLaCodes = new Set([
+  '001',
```

Comment:
We ought to move these LA Codes to the DB soon.

Reply:
thats due to happen soon.  we can cover in next planning session